ksqldb-rest-app/src/test/java/io/confluent/ksql/api/CorsTest.java

```diff
@@ -0,0 +1,289 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import io.confluent.ksql.api.server.ApiServerConfig;
+import io.confluent.ksql.api.server.KSqlCorsHandler;
+import io.confluent.ksql.util.VertxCompletableFuture;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CorsTest extends BaseApiTest {
+
+  private Map<String, Object> config = new HashMap<>();
+
+  private static final String ACCESS_CONTROL_ALLOW_ORIGIN_HEADER = "Access-Control-Allow-Origin";
+  private static final String ACCESS_CONTROL_ALLOW_METHODS_HEADER = "Access-Control-Allow-Methods";
+  private static final String ACCESS_CONTROL_ALLOW_HEADERS_HEADER = "Access-Control-Allow-Headers";
+  private static final String ACCESS_CONTROL_ALLOW_CREDENTIALS_HEADER = "Access-Control-Allow-Credentials";
+
+  private static final String ACCESS_CONTROL_REQUEST_METHOD_HEADER = "Access-Control-Request-Method";
+
+  private static final String DEFAULT_ACCESS_CONTROL_ALLOW_HEADERS = "X-Requested-With,Content-Type,Accept,Origin";
+  private static final String DEFAULT_ACCESS_CONTROL_ALLOW_METHODS = "GET,POST,HEAD";
+
+  @Before
+  public void setUp() {
+
+    vertx = Vertx.vertx();
+    vertx.exceptionHandler(t -> log.error("Unhandled exception in Vert.x", t));
+
+    testEndpoints = new TestEndpoints();
+    setDefaultRowGenerator();
+  }
+
+  @After
+  public void tearDown() {
+    stopClient();
+    stopServer();
+    if (vertx != null) {
+      vertx.close();
+    }
+  }
+
+  @Test
+  public void shouldNotBeCorsResponseIfNoCorsConfigured() throws Exception {
+
+    // Given:
+    init();
+
+    // When
+    HttpResponse<Buffer> response = sendCorsRequest(HttpMethod.POST, "wibble.com");
+
+    // Then:
+    assertThat(response.statusCode(), is(200));
+    assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_ORIGIN_HEADER), is(nullValue()));
+    assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_METHODS_HEADER), is(nullValue()));
+    assertThat(response.getHeader(ACCESS_CONTROL_ALLOW_HEADERS_HEADER), is(nullValue()));
+  }
+
+  @Test
+  public void shouldExcludePath() throws Exception {
+
+    // Given:
+    init();
+    config.put(ApiServerConfig.CORS_ALLOWED_ORIGINS, "wibble.com");
+    KSqlCorsHandler.EXCLUDED_PATH_PREFIXES.add("/query-stream");
```

Comment:
nit: put "wibble.com" and "/query-stream" in constants?

Reply:
ack